crates/usvg-parser/src/lib.rs

```diff
@@ -89,6 +136,46 @@ impl std::fmt::Display for Error {
             Error::ParsingFailed(ref e) => {
                 write!(f, "SVG data parsing failed cause {}", e)
             }
+            Error::UnexpectedEndOfStream => {
+                write!(f, "unexpected end of stream")
+            }
+            Error::UnexpectedData(pos) => {
+                write!(f, "unexpected data at position {}", pos)
+            }
+            Error::InvalidValue => {
+                write!(f, "invalid value")
+            }
+            Error::InvalidIdent => {
+                write!(f, "invalid ident")
+            }
+            Error::InvalidChar(ref chars, pos) => {
+                // Vec<u8> -> Vec<String>
+                let list: Vec<String> = chars
+                    .iter()
+                    .skip(1)
+                    .map(|c| String::from_utf8(vec![*c]).unwrap())
+                    .collect();
+
+                write!(
+                    f,
+                    "expected '{}' not '{}' at position {}",
+                    list.join("', '"),
+                    chars[0] as char,
```

Comment:
Shouldn't it be `chars[pos]`? Also, the square brackets are concerning. It's better to use `.get(...)` since that checks that the position actually exists.

Reply:
This is just due to my ignorance of how to combine the things I brought over from svgtypes. In the original crate the stream parser has it's own set of errors, I wasn't sure how to split it up so I added those errors here.

Reply:
https://github.com/linebender/svgtypes/blob/main/src/error.rs line 64